File: src/cgra_conf_tag.cpp
#include <cgra_conf_tag.h>

ConfTag::ConfTag(bool hasRouter, int numConsts) {
    conf(hasRouter, numConsts);
}

void ConfTag::conf(bool hasRouter, int numConsts) {
    int baseBits = 2;
    if (hasRouter) {
        baseBits = 3;
    }
    this->bits = bitsFunct(baseBits+numConsts);

    this->reset = formatToBinary(0, this->bits);
    this->alu = formatToBinary(1, this->bits);
    this->constant.clear();
    for (int i = 0; i < numConsts; ++i) {
        this->constant.push_back(formatToBinary(2 + i, this->bits));
    }
    this->router = formatToBinary(2 + numConsts, this->bits);
}

File: tests/JsonTst/Python/cgra_conf_tag.py
from src.hw.utils import bits

class ConfTag:
    def __init__(self, hasrouter, num_consts):
        base_bits=2
        if hasrouter:
            base_bits=3
        self.bits = bits(base_bits+num_consts)
        
        self.reset = format(0, '0%db' % self.bits)
        self.alu = format(1, '0%db' % self.bits)
        self.const = [format(2 + i, '0%db' % self.bits) for i in range(num_consts)]
        self.router = format(2 + num_consts, '0%db' % self.bits)